pyqmc/jastrowspin.py

```diff
@@ -305,3 +304,22 @@ def pgradient(self):
         For the derivatives of basis functions, we will have to compute the derivative
         of all the b's and redo the sums, similar to recompute() """
         return {"bcoeff": self._bvalues, "acoeff": self._avalues}
+
+    def u_components(self, r):
+
+        u_onebody = {"up": [], "dn": []}
+        a_value = list(map(lambda x: x.value(None, r), self.a_basis))

```

Comment:
It seems to me that passing None here could cause a crash; it is depending on very specific behavior from the basis functions. Shouldn't we just pass a 3-vector as r?